source/frontend/src/features/mapSideBar/acquisition/common/GenerateForm/GenerateFormContainer.tsx

```diff
@@ -14,16 +22,143 @@ export interface IGenerateFormContainerProps {
 const GenerateFormContainer: React.FunctionComponent<
   React.PropsWithChildren<IGenerateFormContainerProps>
 > = ({ acquisitionFileId, View }) => {
-  const generateLetter = useGenerateLetter();
+  const [fullRecipientsList, setFullRecipientsList] = useState<LetterRecipientModel[]>([]);
+  const { getPersonConcept, getOrganization } = useApiContacts();
+  const {
+    getAcquisitionFile: { execute: getAcquisitionFile },
+  } = useAcquisitionProvider();
+  const [isGenerateLetterModalOpened, openGenerateLetterModal, closeGenerateLetterModal] =
+    useModalManagement();
+  const {
+    getAcquisitionOwners: {
+      execute: retrieveAcquisitionFileOwners,
+      loading: loadingAcquisitionFileOwners,
+    },
+  } = useAcquisitionProvider();
+  const {
+    getAcquisitionInterestHolders: {
+      execute: fetchInterestHolders,
+      loading: loadingInterestHolders,
+    },
+  } = useInterestHolderRepository();
 
-  const generateH0443 = useGenerateH0443();
+  const fetchAllRecipients = useCallback(async () => {
+    const generateRecipientsList: LetterRecipientModel[] = [];
+    const file = await getAcquisitionFile(acquisitionFileId);
+    if (file === undefined) {
+      return;
+    }
 
-  const generateH120 = useGenerateH120();
+    const fileOwnersFetchCall = retrieveAcquisitionFileOwners(acquisitionFileId);
+    const interestHoldersFetchCall = fetchInterestHolders(acquisitionFileId);
+
+    // Add owners and interest holders
+    await Promise.all([fileOwnersFetchCall, interestHoldersFetchCall]).then(
+      ([fileOwners, intHolders]) => {
```

Comment:
This works as-is but could be rewritten to drop the promise.then()

Something like
```tsx
const [fileOwners, intHolders] = await Promise.all([...]);
```

Reply:
Updated.